SKIP approval process (#13)
spinkube/skips: proposals/005-skip-approval-process/README.md

# SKIP 005 - SKIP Approval Process

Summary: A process for approving future SKIPs.

Owner: Caleb Schoepp <caleb.schoepp@fermyon.com>

Impacted Projects:

- [x] spin-operator
- [x] `spin kube` plugin
- [x] runtime-class-manager
- [x] containerd-shim-spin
- [x] Governance
- [ ] Creates a new project

Created: 30/10/2024

Updated: n/a

## Background

SKIPs were designed as a lightweight way to have discussions about changes that are more complex or affect multiple projects. They have worked well for this purpose, but we never codified how we should reach agreement around approving a SKIP. This proposal seeks to change that.

## Proposal

The following conditions must be met for a SKIP to be considered approved:

> - The SKIP has been open for comment for at least two weeks.

Reviewer: I think this condition is not necessary if the other conditions are met, especially being discussed at SpinKube meeting which holds bi-weekly.

Author: Interesting point. My concern would be that someone doesn't have a chance to provide feedback to a SKIP e.g. someone posts a SKIP and the next day chats about it in the community meeting, two people approve right away and it merges — a third maintainer never even gets a reasonable chance to provide feedback or disagree.

Reviewer: Agreed. Let's keep it there

> - The SKIP has been open for comment for at least two weeks.

Reviewer 2: Do we need "and any objecting comments are resolved"? For example, this skip i may has been open for 2 weeks but i dont think it should be merged: https://github.com/spinkube/skips/pull/14

Author: I think that is a good addition. I'll add it.

> - The SKIP has been open for comment for at least two weeks and any objecting comments are resolved.

Reviewer 2: "Objecting" is subjective. IMO this should be all comments
```suggestion
- The SKIP has been open for comment for at least two weeks and all comments are resolved.
```

Author: Good call. I'll followup with this change.
